### ISSUE TITLE
GPU is not accelerated

### ISSUE THREAD

**Issue by stromyu520** [roboflow/supervision#1909]
### Search before asking

- [x] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

The console stream is logged into /root/sg_logs/console.log
[2025-07-23 19:06:07] INFO - crash_tips_setup.py - Crash tips is enabled. You can set your environment variable to CRASH_HANDLER=FALSE to disable it
[2025-07-23 19:06:08] WARNING - __init__.py - Failed to import pytorch_quantization
[2025-07-23 19:06:10] WARNING - calibrator.py - Failed to import pytorch_quantization
[2025-07-23 19:06:10] WARNING - export.py - Failed to import pytorch_quantization
[2025-07-23 19:06:10] WARNING - selective_quantization_utils.py - Failed to import pytorch_quantization
[2025-07-23 19:06:10] INFO - checkpoint_utils.py - License Notification: YOLO-NAS pre-trained weights are subjected to the specific license terms and conditions detailed in
https://github.com/Deci-AI/super-gradients/blob/master/LICENSE.YOLONAS.md
By downloading the pre-trained weight files you agree to comply with these terms.
[2025-07-23 19:06:11] INFO - checkpoint_utils.py - Successfully loaded pretrained weights for architecture yolo_nas_l
FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
[2025-07-23 19:06:11] INFO - pipelines.py - Fusing some of the model's layers. If this takes too much memory, you can deactivate it by setting `fuse_model=False`
[2025-07-23 19:06:12] INFO - pipelines.py - Fusing some of the model's layers. If this takes too much memory, you can deactivate it by setting `fuse_model=False`
[2025-07-23 19:06:12] INFO - pipelines.py - Fusing some of the model's layers. If this takes too much memory, you can deactivate it by setting `fuse_model=False`
[2025-07-23 19:06:13] INFO - pipelines.py - Fusing some of the model's layers. If this takes too much memory, you can deactivate it by setting `fuse_model=False`
[2025-07-23 19:06:14] INFO - pipelines.py - Fusing some of the model's layers. If this takes too much memory, you can deactivate it by setting `fuse_model=False`
[2025-07-23 19:06:14] INFO - pipelines.py - Fusing some of the model's layers. If this takes too much memory, you can deactivate it by setting `fuse_model=False`
[2025-07-23 19:06:15] INFO - pipelines.py - Fusing some of the model's layers. If this takes too much memory, you can deactivate it by setting `fuse_model=False`
[2025-07-23 19:06:16] INFO - pipelines.py - Fusing some of the model's layers. If this takes too much memory, you can deactivate it by setting `fuse_model=False`
[2025-07-23 19:06:16] INFO - pipelines.py - Fusing some of the model's layers. If this takes too much memory, you can deactivate it by setting `fuse_model=False`

**Comment by soumik12345** [comment]
Hi @stromyu520, can you please clarify what issue you are facing and attach a minimal reproducible code snippet?

**Comment by soumik12345** [comment]
Hi @stromyu520, this seems like an issue with the `super_gradients` library, I recommend that you post your issue on https://github.com/Deci-AI/super-gradients

**Comment by SkalskiP** [comment]
Hi @stromyu520 👋🏻 based on this example [notebook](https://colab.research.google.com/github/roboflow-ai/notebooks/blob/main/notebooks/train-yolo-nas-on-custom-dataset.ipynb), I suggest trying to update `yolo_nas_example.py` script.

- Before:

    ```python
    model = models.get(Models.YOLO_NAS_L, pretrained_weights="coco")
    ```

- After:

    ```python
    import torch

    DEVICE = 'cuda' if torch.cuda.is_available() else "cpu"
    model = models.get(Models.YOLO_NAS_L, pretrained_weights="coco").to(DEVICE)
    ```

**Comment by SkalskiP** [comment]
@stromyu520 I'm closing the issue. Feel free to reopen it, if you'll have more questions.
